pkg/zfs/zfs_util.go

```diff
@@ -127,9 +127,41 @@ func buildCloneCreateArgs(vol *apis.ZFSVolume) []string {
 	ZFSVolArg = append(ZFSVolArg, ZFSCloneArg)
 
 	if vol.Spec.VolumeType == VOLTYPE_DATASET {
+		if len(vol.Spec.Capacity) != 0 {
+			quotaProperty := "quota=" + vol.Spec.Capacity
+			ZFSVolArg = append(ZFSVolArg, "-o", quotaProperty)
+		}
+		if len(vol.Spec.RecordSize) != 0 {
+			recordsizeProperty := "recordsize=" + vol.Spec.RecordSize
+			ZFSVolArg = append(ZFSVolArg, "-o", recordsizeProperty)
+		}
+		if vol.Spec.ThinProvision == "no" {
+			reservationProperty := "reservation=" + vol.Spec.Capacity
```

Comment:
Is this the right spec value?

Reply:
yes it is.

 reservation=size|none
       The minimum amount of space guaranteed to a dataset and its descendants.  When the amount of space used is below this value, the dataset is treated as if it
       were taking up the amount of space specified by its reservation.  Reservations are accounted for in the parent datasets' space used, and count against the
       parent datasets' quotas and reservations.

       This property can also be referred to by its shortened column name, reserv.

Reply:
cool. trying to understand this a little more. 

Say: 

PV1 = 5G ( quota and reservation will be set to 5G)
PV1Clone1 = 5G (quota and reservation will be set to 5G) 
PV1Clone2 = 5G (quota and reservation will be set to 5G) 

Now, PV1Clone1 and PVClone2 will be considered as children of PV1. Will the PV1 run out of space if PV1Clone1 and PV1Clone2 start consuming more space?

Reply:
clone is not a child of original volume. It will have it's separate quota and reservation.